service/host.go

```diff
@@ -247,19 +270,23 @@ func (s *host) Start(waitForShutdown bool) (<-chan Exit, <-chan struct{}, error)
 				if _, err := s.shutdown(e, "", nil); err != nil {
 					s.Logger().Error("Unable to shut down modules", "initialError", e, "shutdownError", err)
 				}
-				return errChan, readyCh, e
+				return Control{
+					ExitChan:     errChan,
+					ReadyChan:    readyCh,
+					ServiceError: e,
```

Comment:
`e` is used to range over errors, but we exit with the first cycle. We should log all the errors, then return the concatenation of all the errors with a wrap saying `failed to start the modules because: a b and c`